common/src/main/java/whocraft/tardis_refined/common/tardis/themes/DesktopTheme.java

```diff
@@ -4,25 +4,36 @@
 import net.minecraft.resources.ResourceLocation;
 import whocraft.tardis_refined.constants.ModMessages;
 
+import java.util.Optional;
+
 
 public class DesktopTheme {
 
     private final ResourceLocation uiTexture;
     public boolean availableByDefault = false;
-    public String id = "";
+    public ResourceLocation identifier;
+    public String name = "";
     public ResourceLocation location;
 
 
 
-    public DesktopTheme(String id, ResourceLocation location, boolean availableByDefault) {
-        this.id = id;
+    public DesktopTheme(ResourceLocation id, ResourceLocation location, boolean availableByDefault) {
+        this.identifier = id;
         this.availableByDefault = availableByDefault;
         this.location = location;
-        this.uiTexture = new ResourceLocation(location.getNamespace(), "textures/ui/interiors/" + id + ".png");
+        this.uiTexture = new ResourceLocation(location.getNamespace(), "textures/ui/interiors/" + id.getPath().toString() + ".png");
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
     }
 
     public String getTranslationKey(){
-        return ModMessages.desktop(id);
+        return ModMessages.desktop(identifier.getPath());
```

Comment:
Do we have much need for getTranslationKey with the tellraw stuff now on the scene? We should probably remove it from all 3 datapack entry types